### PR TITLE
Design Panel: Fix transparent color edit

### LINKED ISSUE #5369: Fully transparent color looks weird
## Bug Description

If you have a fully transparent solid color anywhere (text color, border color, text background color, shape color), the result is really weird in the preview. See screenshots below

## Expected Behaviour

* The preview should somehow reflect the transparent color

## Steps to Reproduce

1. Add a shape
1. Change the color opacity to 0%
1. Observe weird behavior

## Screenshots

disappearing color and opacity

## Additional Context

- Plugin Version:
- Operating System:
- Browser:

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria

1. Show the non-transparent color in color preview
1. Write the actual hex in the editable text field
1. Show the opacity input with 0% as the value

It should work similarly to a fully-transparent gradient:
Screen Shot 2020-11-20 at 12 14 22

### PR DESCRIPTION
**Pull request by zidianlyu**
## Summary
Fix transparent color edit under design panel.

## To-do

N/A
## User-facing changes

The color editor should persist when user change the opacity to 0%

## Testing Instructions

The color editor should persist when user change the opacity to 0%

---

Fixes #5369

### CODE REVIEW

File: assets/src/edit-story/components/form/color/color.js [hunk 1]
```diff
@@ -70,9 +70,10 @@ function ColorInput({
         colorPickerActions={colorPickerActions}
         changedStyle={changedStyle}
       />
-      {hasOpacity && (
+      {/* {hasOpacity && (
         <OpacityPreview value={value} onChange={handleOpacityChange} />
-      )}
+      )} */}
```
Comment (swissspidy) [inline assets/src/edit-story/components/form/color/color.js]: Debugging leftover? Should remove stuff instead of commenting things out

**Comment by swissspidy** [comment]
> The color editor should persist when user change the opacity to 0%

Could you perhaps add before/after screenshots to the PR description?

File: assets/src/edit-story/components/form/color/color.js (hunk 1 above)
Comment (zidianlyu) [reply assets/src/edit-story/components/form/color/color.js]: Good catch, bring back!

**Comment by zidianlyu** [comment]
Screenshots:

Before the fix:
Screen Shot 2020-12-21 at 9 06 00 AM


After the fix:
Screen Shot 2020-12-21 at 9 04 39 AM

**Review by swissspidy — APPROVED** [review]

File: assets/src/edit-story/components/form/color/getPreviewStyle.js [hunk 2]
```diff
@@ -32,12 +32,8 @@ function getPreviewStyle(pattern) {
...
   const {
-    color: { r, g, b, a },
+    color: { r, g, b },
   } = pattern;
-  // If opacity is 0, no extra styling needed:
-  if (a === 0) {
-    return {};
-  }
 
   // Otherwise create color, but with full opacity
```
Comment (dreamofabear) [inline assets/src/edit-story/components/form/color/getPreviewStyle.js]: This comment ("otherwise") doesn't make sense anymore.

**Review by dreamofabear — APPROVED** [review]

File: assets/src/edit-story/components/form/color/getPreviewStyle.js (hunk 2 above)
Comment (zidianlyu) [reply assets/src/edit-story/components/form/color/getPreviewStyle.js]: cleaned up

**Pull request merged**